.circleci/config.yml

```diff
@@ -13,17 +13,27 @@ jobs:
       e:
         type: executor
         default: php_8_0
+      cv:
+        description: "composer's version"
+        type: integer
+        default: 2
     executor: << parameters.e >>
     steps:
       - checkout
+      - run:
+          description: "Setup composer v<< parameters.cv >>"
+          command: >
+            sudo composer self-update
```

Comment:
I was able to reproduce the memory error in a local CI build (by running the following:
```
circleci config process .circleci/config.yml > .circleci/local-config.yml && circleci local execute -c .circleci/local-config.yml --job 'build_and_test-2'
```

The error stopped occurring when I remove the `sudo composer self-update ...`. Do you think removing that command would be OK?

Separately, the `v1` pattern in the key names was originally intended for forced invalidation of stale caches (e.g. the config would be change to `v2`, 'v3', 'vn+1'). I don't see a problem with converting this to be used for php version - if we need to later, we can always add back a new `vX` to the key names.

Reply:
thanks for the review,

> The error stopped occurring when I remove the sudo composer self-update .... Do you think removing that command would be OK?

removing it would completely missing the point of this PR...
this downgrades composer to version1 when needed. This allows to test compoer1 in the first place.
see #24 for details.

Reply:
@jkowalleck Could you merge in the changes from the `main` branch, and use the command I posted above to test CI locally? If it works, please push the merged changes up into this PR. (Fingers crossed). ;)